src/components/bills/index.tsx

```diff
@@ -44,11 +43,40 @@ export default function UploadBillForm() {
     Boat: "",
     Type: "",
     Amount: "",
-    pdfFile: {} as File,
+    pdfFiles: [],
   };
 
   const [data, setData] = useState(initialState);
   var boatInfo = {} as Boat;
+
+  const convertImagesToSeparatePdfs = async (images: File[]): Promise<File[]> => {
+    try {
+      const pdfFiles: File[] = [];
+
+      for (const image of images) {
+        const pdf = new jsPDF();
+        const imageData = await new Promise<string>((resolve, reject) => {
+          const reader = new FileReader();
+          reader.onload = (e) => resolve(e.target?.result as string);
+          reader.onerror = reject;
+          reader.readAsDataURL(image);
+        });
+
+        pdf.addImage(imageData, "JPEG", 10, 10, 180, 240);
+        const pdfBlob = pdf.output("blob");
+        const pdfFile = new File([pdfBlob], `${image.name.split(".")[0]}.pdf`, {
+          type: "application/pdf",
+        });
+        pdfFiles.push(pdfFile);
+      }
+
+      return pdfFiles;
+    } catch (error) {
+      console.error("Error converting images to separate PDFs:", error);
+      return [];
+    }
+  };
```

Comment:
This shall be on the input (with a property fi we needs originals or PDFS?)

Or a second layer component multipleFilesToPdf adding this functionality